clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -461,6 +507,10 @@ public void close(Duration timeout) {
 
     @Override
     public void wakeup() {
+        this.shouldWakeup.set(true);
+        if (this.activeFutures.get() != null) {
+            this.activeFutures.getAndSet(null).completeExceptionally(new WakeupException());
```

Comment:
This does not seem safe. If there are concurrent calls to `getAndSet`, we might see NPE?